lib/jekyll/static_file.rb

```diff
@@ -104,7 +100,7 @@ def to_liquid
     end
 
     def data
-      @data ||= {}
+      @data ||= @site.frontmatter_defaults.all(relative_path, type)
```

Comment:
Why not resort to existing method `:defaults` in [`StaticFile`](https://github.com/jekyll/jekyll/blob/v3.5.0/lib/jekyll/static_file.rb#L144-L148)

```ruby
def data
  @data ||= defaults
end
```